backend/src/bundles/user-details/user-details.service.ts

```diff
@@ -107,15 +110,35 @@ class UserDetailsService implements Service {
         throw new Error(ErrorMessage.NOT_IMPLEMENTED);
     }
 
-    public searchUsers(
+    public async searchUsers(
         searchData: UserDetailsSearchUsersRequestDto,
-    ): Promise<UserDetailsEntity[]> {
+    ): Promise<UserDetailsProperties[]> {
         const preparedData = mapQueryValuesToArrays(searchData, [
             'searchValue',
             'sortBy',
+            'searchType',
+            'searchStringType',
         ]);
 
-        return this.userDetailsRepository.searchUsers(preparedData);
+        const filteredUsers = await this.userDetailsRepository.searchUsers(
+            preparedData,
+        );
+
+        const userPromises = filteredUsers.map(async (user) => {
+            const userDetails = user.toObject();
+            const userDetailsId = userDetails.id as string;
+
+            const userHardSkills = await getUserHardSkills(userDetailsId);
+
+            const userBadges = await getUserBadges(userDetailsId);
+
+            userDetails.hardSkills = userHardSkills;
+            userDetails.badges = userBadges;
+
+            return userDetails;
+        });
```

Comment:
Why we can't fetch users hard skills and badges using withGraphFetched method for example and in this case we join data using sql raw query and it much better than async map with Promise.all